coalib/misc/Caching.py

```diff
@@ -0,0 +1,81 @@
+import calendar
```

Comment:
Uh oh, I forgot to change the commit message. I'll do it along with the changes from review comments, if any.